### PR TITLE
Fix failure of cross-compiling for ffmpeg due to use inappropriate strip.

### PR DESCRIPTION
**Pull request by Jamlys**
Fix failure of cross-compiling for ffmpeg due to use inappropriate strip

**Describe the pull request**

- #### What does your PR fix?
  Fix failure of cross-compiling for ffmpeg due to use inappropriate strip

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
linux,No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Basically.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
I am still working on this PR.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/ffmpeg/portfile.cmake [hunk 1]
```diff
@@ -458,6 +460,11 @@ endif()
 
 set(OPTIONS_CROSS " --enable-cross-compile")
 
+# ffmpeg needs --cross-prefix option to use appropriate tools for cross-compiling.
+if(VCPKG_DETECTED_CMAKE_C_COMPILER MATCHES "(/.+)gcc$" AND CMAKE_MATCH_1)
```
Comment (dg0yt) [inline ports/ffmpeg/portfile.cmake:464]: I guess it should be:
~~~cmake
if(VCPKG_DETECTED_CMAKE_C_COMPILER MATCHES "/(.+)gcc$")
~~~
(Moved `/`. `CMAKE_MATCH_1` is implied.)

Comment (Jamlys) [reply ports/ffmpeg/portfile.cmake:464]: Yep. I took this line from vcpkg_configure_make.cmake(377) while I was tracing the problem and lazily remove the redundant condition. My fault.
